### PR TITLE
Orchestration post actions support

### PR DESCRIPTION
**Pull request by guderkar**

### CODE REVIEW

File: src/featurestorebundle/orchestration/PostActionsRunner.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+        for post_action_service, post_action in zip(post_action_services, self.__post_actions):
+            self.__logger.info(f"Running post action {post_action}")
+
+            post_action_service.run()
+
+            self.__logger.info(f"Post action {post_action} done")
+
+    def __get_post_action_services(self) -> List[PostActionInterface]:
+        post_actions_unified = [post_action[1:] if post_action.startswith("@") else post_action for post_action in self.__post_actions]
```
Comment (ds-lukaslangr) [inline src/featurestorebundle/orchestration/PostActionsRunner.py:29]: the post actions can start without @?

File: src/featurestorebundle/orchestration/PostActionsRunner.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    def __init__(self, logger: Logger, post_actions: Box):
+        self.__logger = logger
+        self.__post_actions = post_actions
+
+    def run(self):
+        post_action_services = self.__get_post_action_services()
+
+        self.__validate_post_action_services(post_action_services)
+
+        if len(self.__post_actions) == 0:
```
Comment (ds-lukaslangr) [inline src/featurestorebundle/orchestration/PostActionsRunner.py:18]: if not self.__post_actions

File: src/featurestorebundle/orchestration/PostActionsRunner.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            self.__logger.info(f"Post action {post_action} done")
+
+    def __get_post_action_services(self) -> List[PostActionInterface]:
+        post_actions_unified = [post_action[1:] if post_action.startswith("@") else post_action for post_action in self.__post_actions]
+
+        return [ContainerManager.get_container().get(post_action) for post_action in post_actions_unified]
+
+    def __validate_post_action_services(self, post_action_services: List[PostActionInterface]):
+        for post_action_service in post_action_services:
+            if not isinstance(post_action_service, PostActionInterface) or not hasattr(post_action_service, "run"):
```
Comment (ds-lukaslangr) [inline src/featurestorebundle/orchestration/PostActionsRunner.py:35]: what if `run` isn't a method but a property?

Comment (ds-lukaslangr) [reply src/featurestorebundle/orchestration/PostActionsRunner.py:35]: also doesn't every PostActionInterface have `run` from the parent?

File: src/featurestorebundle/orchestration/PostActionsRunner.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    def run(self):
+        post_action_services = self.__get_post_action_services()
+
+        self.__validate_post_action_services(post_action_services)
+
+        if len(self.__post_actions) == 0:
+            self.__logger.info("No post actions")
+
+        for post_action_service, post_action in zip(post_action_services, self.__post_actions):
```
Comment (ds-lukaslangr) [inline src/featurestorebundle/orchestration/PostActionsRunner.py:21]: what's the difference between `post_action_services` and `self.__post_actions`?

**Review by ds-lukaslangr — CHANGES_REQUESTED** [review]

File: src/featurestorebundle/orchestration/PostActionsRunner.py (hunk 3 above)
Comment (guderkar) [reply src/featurestorebundle/orchestration/PostActionsRunner.py:35]: yes but it should raise not implemented if not implemented due to @abc decorator

File: src/featurestorebundle/orchestration/PostActionsRunner.py (hunk 4 above)
Comment (guderkar) [reply src/featurestorebundle/orchestration/PostActionsRunner.py]: renamed to post_action_service_identifier

File: src/featurestorebundle/orchestration/PostActionsRunner.py (hunk 3 above)
Comment (guderkar) [reply src/featurestorebundle/orchestration/PostActionsRunner.py]: So basically the second condition is useless

**Pull request merged**